packages/core/src/components/advanced-select/advanced-select.tsx

```diff
@@ -0,0 +1,160 @@
+   * The active option. By object reference.
+   */
+  @Prop()
+  active?: AdvancedSelectOption<any>;
+
+  /**
+   * An extra text for the active option. Only visible in the list of options.
+   */
+  @Prop()
+  activeHint?: string;
+
+  /**
+   * The open state of the options list.
+   */
+  @Prop()
+  open: boolean = false;
+
+  /**
+   * Emitted when user clicks the select.
+   */
+  @Event({ bubbles: false })
+  dsoClick!: EventEmitter<AdvancedSelectClickEvent>;
+
+  /**
+   * Emitted when user clicks an option
+   */
+  @Event({ bubbles: false })
+  dsoOptionClick!: EventEmitter<AdvancedSelectOptionClickEvent<any>>;
+
+  /**
+   * Emitted when user clicks a redirect link.
+   */
+  @Event({ bubbles: false })
+  dsoRedirectClick!: EventEmitter<AdvancedSelectRedirectClickEvent>;
+
+  private handleClick = (event: MouseEvent) => {
+    this.dsoClick.emit({ originalEvent: event });
+  };
+
+  private handleOptionClick = (event: MouseEvent, option: AdvancedSelectOption<any>) => {
+    this.dsoOptionClick.emit({ originalEvent: event, option });
+  };
+
+  private handleRedirectClick = (event: MouseEvent, redirect: AdvancedSelectGroupRedirect) => {
+    this.dsoRedirectClick.emit({ originalEvent: event, redirect });
+  };
+
+  render() {
+    return (
+      <div>
+        <button class={{ "active-option": true, open: this.open }} type="button" onClick={this.handleClick}>
+          <span class="active-option-label">{this.active?.label ?? "Selecteer een optie"}</span>
+          <span class="active-option-aside">
+            {this.options.some((option) => "summaryCounter" in option && option?.summaryCounter) ? (
+              <span class="badges">
+                {this.options
+                  .filter((option) => "summaryCounter" in option && option?.summaryCounter)
+                  .map((option) => {
+                    if ("options" in option) {
+                      return <dso-badge status={option.variant ?? "outline"}>{option.options.length}</dso-badge>;
+                    }
+                  })}
+              </span>
+            ) : (
+              ""
+            )}
+            <dso-icon icon="caret-down"></dso-icon>
+          </span>
+        </button>
+        <div class={{ "groups-container": true, open: this.open }}>
+          <ul class="groups">
+            {this.options.map((optionsOrGroup) => {
+              if ("options" in optionsOrGroup) {
+                return (
+                  <li class={{ group: true, [`group-${optionsOrGroup.variant}`]: !!optionsOrGroup.variant }}>
+                    {optionsOrGroup.label && <p class="group-label">{optionsOrGroup.label}</p>}
+                    <ul class="options">
+                      {optionsOrGroup.options.map((option) => (
+                        <OptionElement
+                          option={option}
+                          active={this.active}
+                          activeHint={this.activeHint}
+                          cb={this.handleOptionClick}
```

Comment:
Liever geen afkortingen zoals cb of bg gebruiken. Je moet maar zo denken: Je leest het vaker dan dat je het typt